### ISSUE TITLE
Undefined variable: b:node_root

### ISSUE THREAD

**Issue by FuzzOli87** [carlitux/deoplete-ternjs#17]
Hello!

I've been using Deoplete with Deoplete-tern. There has been a consistent error message when I'm working on some projects. The projects are those that use the require() import instead of the new import statement. Is there any chance this is an issue related to this plugin?

```
[deoplete] Traceback (most recent call last):
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 47, in completion_begin
[deoplete]     complete_position, candidates = self.gather_candidates(context)
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 81, in gather_candidates
[deoplete]     results = self.gather_results(context)
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/deoplete.py", line 132, in gather_results
[deoplete]     context['candidates'] = source.gather_candidates(context)
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 32, in gather_candidates
[deoplete]     self.__make_cache(context)
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 43, in __make_cache
[deoplete]     for filename in self.__get_tagfiles():
[deoplete]   File "/Users/daniel.olivares/.nvim/plugged/deoplete.nvim/rplugin/python3/deoplete/sources/tag.py", line 55, in __get_tagfiles
[deoplete]     'exists', '*neoinclude#include#get_tag_files') else []
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/api/nvim.py", line 213, in call
[deoplete]     return self.request('vim_call_function', name, args, **kwargs)
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/api/nvim.py", line 129, in request
[deoplete]     res = self._session.request(name, *args, **kwargs)
[deoplete]   File "/usr/local/lib/python3.5/site-packages/neovim/msgpack_rpc/session.py", line 98, in request
[deoplete]     raise self.error_wrapper(err)
[deoplete] neovim.api.nvim.NvimError: b'Vim(return):E121: Undefined variable: b:node_root'
[deoplete] An error has occurred. Please execute :messages command.
```

**Comment by carlitux** [comment]
It is related in some way... I am getting also this some times I think is on gather_results but I couldn't find how this happens.

But reading your traceback looks like now it is related with tag files when tries to gather_results on it.

```bash
[deoplete]     for filename in self.__get_tagfiles():
```

**Comment by FuzzOli87** [comment]
@carlitux 
Do you by any chance have [neoinclude.vim](https://github.com/Shougo/neoinclude.vim) installed?

**Comment by carlitux** [comment]
@FuzzOli87 yes I have installed.

**Comment by FuzzOli87** [comment]
@carlitux 
That is the source of the issue. I'm not exactly sure what the plugin is meant to do, I don't even remember why I installed it. I looked it up in the deoplete docs and it was a recommended install so I added it. I know, lazy.

Anyways, I removed it and it stopped the errors. Not sure what the cause is but I might go back and investigate further. I don't get those errors when I'm using ES6 modules though, at least not that I remember.
